### PR TITLE
refactor: remove usage of unstable `generic_const_exprs` in starky

### PR DESCRIPTION
**Pull request by bingcicle**
This PR mostly lifts https://github.com/0xPolygonZero/plonky2/pull/1246 down into the lib level into the starky crate, getting rid of the unstable `generic_const_exprs` feature from the crate entirely. This should work towards the https://github.com/0xPolygonZero/plonky2/issues/417 as well.

This would probably warrant a minor version bump in the starky crate to 0.2.0, since this would be an API change for anyone relying on starky.

### CODE REVIEW

File: starky/src/lib.rs [hunk 1]
```diff
@@ -1,7 +1,6 @@
 #![allow(incomplete_features)]
```
Comment (Nashtare) [inline starky/src/lib.rs:1]:
```suggestion
```

I believe this can also be removed now :)

Comment (bingcicle) [reply starky/src/lib.rs]: You're right! Thanks for the catch

File: starky/src/evaluation_frame.rs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    const COLUMNS: usize;
+    const PUBLIC_INPUTS: usize;
+
+    /// Returns the local values (i.e. current row) for this evaluation frame.
+    fn get_local_values(&self) -> &[T];
+    /// Returns the next values (i.e. next row) for this evaluation frame.
+    fn get_next_values(&self) -> &[T];
+
+    fn get_public_inputs(&self) -> &[U];
+    /// Outputs a new evaluation frame from the provided local and next values.
```
Comment (Nashtare) [inline starky/src/evaluation_frame.rs:19]:
```suggestion
    fn get_public_inputs(&self) -> &[U];
    
    /// Outputs a new evaluation frame from the provided local and next values.
```

minor nit, but I'd add a separation here.

**Review by Nashtare — APPROVED** [review]
LGTM! Thanks for doing this :)

**Pull request merged**